instagram/src/components/authentication/withAuthenticate.js

```diff
@@ -0,0 +1,40 @@
+// const withUpperCase = SomeComp => props => {
+//   const upperCaseText = props.text.toUpperCase();
+//   return <SomeComp text={upperCaseText} />;
+// };
+import React from "react"
+
+
+const withAuthenticate = PostsPage => LoginPage => 
+  class extends React.Component {
+    constructor () {
+      super ();
+      this.state = {
+        loggedIn: false
+      }
+      // console.log('construction')
+    };
+  
+    componentDidMount() {
+      if (localStorage.getItem("username")) {
+        this.setState ({
+          loggedIn: true
+        })
+      } else {
+        this.setState ({
+          loggedIn: false
+        })
+      }
+    }
+    
+    render() {
+      if (this.state.loggedIn === true) {
```

Comment:
You also could just write:

if (this.state.loggedIn) {
}

It will be basically run if true, and won't run if loggedIn in false. so it reads like: if true, do this.